garage/tf/algos/ddpg.py

```diff
@@ -36,359 +27,87 @@ class DDPG(RLAlgorithm):
-                                          np.mean(self.success_history))
-
-                # Uncomment the following if you want to calculate the average
-                # in each epoch, better uncomment when self.use_her is True
-                # episode_rewards = []
-                # episode_actor_losses = []
-                # episode_critic_losses = []
-                # epoch_ys = []
-                # epoch_qs = []
-
-            logger.dump_tabular(with_prefix=False)
-            logger.pop_prefix()
-            if self.plot:
-                self.plotter.update_plot(self.actor, self.n_rollout_steps)
-                if self.pause_for_plot:
-                    input("Plotting evaluation run: Press Enter to "
-                          "continue...")
-
-        if self.plot:
-            self.plotter.close()
-        if created_session:
-            sess.close()
-
-    def _initialize(self):
+    def init_opt(self):
         with tf.name_scope(self.name, "DDPG"):
-            with tf.name_scope("setup_networks"):
-                """Set up the actor, critic and target network."""
-                # Set up the actor and critic network
-                self.actor._build_net(trainable=True)
-                self.critic._build_net(trainable=True)
-
-                # Create target actor and critic network
-                target_actor = copy(self.actor)
-                target_critic = copy(self.critic)
-
-                # Set up the target network
-                target_actor.name = "TargetActor"
-                target_actor._build_net(trainable=False)
-                target_critic.name = "TargetCritic"
-                target_critic._build_net(trainable=False)
-
-            input_shapes = dims_to_shapes(self.input_dims)
-
-            # Initialize replay buffer
-            if self.use_her:
-                buffer_shapes = {
-                    key: (self.n_rollout_steps + 1
-                          if key == "observation" or key == "achieved_goal"
-                          else self.n_rollout_steps, *input_shapes[key])
-                    for key, val in input_shapes.items()
-                }
-
-                replay_buffer = HerReplayBuffer(
-                    buffer_shapes=buffer_shapes,
-                    size_in_transitions=self.replay_buffer_size,
-                    time_horizon=self.n_rollout_steps,
-                    sample_transitions=make_her_sample(
-                        self.replay_k, self.env.compute_reward))
-            else:
-                replay_buffer = ReplayBuffer(
-                    buffer_shapes=input_shapes,
-                    max_buffer_size=self.replay_buffer_size)
+            # Create target policy and qf network
+            self.target_policy_f_prob_online, _, _ = self.policy._build_net(
+                trainable=False, name="target_policy")
+            self.target_qf_f_prob_online, _, _, _ = self.baseline._build_net(
```

Comment:
There is no baseline in ddpg.

Reply:
To be less smug, a baseline is a value function (estimates vf: state --> \E{sum of discounted rewards}), while a Q-Function estimates  (qf: (state, action) --> \E{sum of discounted rewards})

Reply:
I didn't get a better idea on how to address this.
For now, OffPolicyBatchPolopt and OnPolicyBatchPolopt both inherit from BatchPolopt, `BatchPolopt.log_diagnostics` needs to use `self.baseline`, so I used policy and baseline in BatchPolopt's constructor. 
I can override `log_diagnostics` in OffPolicyBatchPolopt